src/backend/executor/process.ts

```diff
@@ -270,20 +272,26 @@ export class ExecutorManager implements Disposable {
         this.clearQueue();
     }
 
-    private updateStatus(status: ProcessStatus) {
+    private updateStatus([status, process]: [ProcessStatus, ScheduledProcess]) {
         switch (status) {
         case ProcessStatus.removed:
-            this._processStatusChange.fire(status);
+            this._processStatusChange.fire([status, process]);
             break;
         case ProcessStatus.running:
-            this._processStatusChange.fire(ProcessStatus.running);
+        case ProcessStatus.queued:
```

Comment:
Is the above fallthrough on line 280 intentional?

Reply:
It is, it's not really a fallthrough, just multiple values executing the same case with no extra code.